analyticsservice/analytics_test.go

```diff
@@ -0,0 +1,50 @@
+package analyticsservice
+
+import "testing"
+
+func Test_convertInterfaceToString(t *testing.T) {
+	testCases := map[string]struct {
+		input    interface{}
+		expected string
+	}{
+		"Given I input a string": {
```

Comment:
should we add some nils? check it handles it ok?

Reply:
Will add a test for nils